includes/models/model.llms.transaction.php

```diff
@@ -69,30 +69,97 @@ class LLMS_Transaction extends LLMS_Post_Model {
 	);
 
 	/**
-	 * Determine if the transaction can be refunded
-	 * Status must not be "failed" and total refunded amount must be less than order amount
+	 * Determines if the transaction can be refunded.
 	 *
-	 * @return   boolean
-	 * @since    3.0.0
-	 * @version  3.0.0
+	 * Status must not be "failed" and total refunded amount must be less than order amount.
+	 *
+	 * @since 3.0.0
+	 * @since [version] Made the return value filterable via the `llms_transaction_can_be_refunded` hook.
+	 *
+	 * @return boolean
 	 */
 	public function can_be_refunded() {
-		$status = $this->get( 'status' );
-		// Can't refund failed or pending transactions.
-		if ( 'llms-txn-failed' === $status || 'llms-txn-pending' === $status ) {
-			return false;
+
+		$can_be_refunded = true;
+
+		if ( in_array( $this->get( 'status' ), array( 'llms-txn-failed', 'llms-txn-pending' ), true ) ) {
+			$can_be_refunded = false;
 		} elseif ( $this->get_refundable_amount( array(), 'float' ) <= 0 ) {
-			return false;
+			$can_be_refunded = false;
+		}
+
+		/**
+		 * Filters whether or not a transaction can be refunded.
+		 *
+		 * @since [version]
+		 *
+		 * @param boolean          $can_be_refunded Whether the transaction can be refunded.
+		 * @param LLMS_Transaction $transaction     The transaction object.
+		 */
+		return apply_filters( 'llms_transaction_can_be_refunded', $can_be_refunded, $this );
+
+	}
+
+	/**
+	 * Generates a refund ID based on the refund method.
+	 *
+	 * When manually processing a refund an ID is generated, if processing via a gateway the data
+	 * is passed to the {@see LLMS_Transaction::process_refund_via_gateway()} and ultimately passed
+	 * to the gateway for processing (via the gateway's API). For custom methods processing is handled
+	 * via the `llms_{$method}_refund_id` filter.
+	 *
+	 * @since [version]
+	 *
+	 * @param string $method Refund processing method ID.
+	 * @param float  $amount The amount to refund.
+	 * @param string $note   Refund notes.
+	 * @return string|boolean|WP_Error Returns the generated refund ID string or an error object. If a falsy value
+	 *                                 is returned the refund processing with fail with a generic error message.
```

Comment:
I'm not sure what the 2nd sentence means. Should it be?
```suggestion
	 * @return string|boolean|WP_Error Returns the generated refund ID string or an error object. If a falsy value
	 *                                 is returned the refund processing will fail with a generic error message.
```